Thekla fails building for Win32 with mingw-w64

It gives these error:

```
[ 59%] /usr/bin/i686-w64-mingw32-g++ -o thirdparty/thekla_atlas/nvcore/FileSystem.windows.opt.tools.32.o -c -std=c++11 -g1 -O2 -DDEBUG_ENABLED -DWINDOWS_ENABLED -mwindows -DOPENGL_ENABLED -DRTAUDIO_ENABLED -DWASAPI_ENABLED -DWINVER=0x0601 -D_WIN32_WINNT=0x0601 -w -DZSTD_STATIC_LINKING_ONLY -DNV_OS_MINGW -DNV_CC_GNUC -DPOSH_COMPILER_GCC -U__STRICT_ANSI__ -fopenmp -DMINGW_ENABLED -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -DGLAD_ENABLED -DGLES_OVER_GL -DMODULE_BULLET_ENABLED -DMODULE_DDS_ENABLED -DMODULE_ENET_ENABLED -DMODULE_ETC_ENABLED -DMODULE_FREETYPE_ENABLED -DMODULE_GDNATIVE_ENABLED -DMODULE_GDSCRIPT_ENABLED -DMODULE_GRIDMAP_ENABLED -DMODULE_HDR_ENABLED -DMODULE_JPG_ENABLED -DMODULE_MOBILE_VR_ENABLED -DMODULE_OGG_ENABLED -DMODULE_OPENSSL_ENABLED -DMODULE_OPUS_ENABLED -DMODULE_PBM_ENABLED -DMODULE_PVR_ENABLED -DMODULE_RECAST_ENABLED -DMODULE_REGEX_ENABLED -DMODULE_SQUISH_ENABLED -DMODULE_STB_VORBIS_ENABLED -DMODULE_SVG_ENABLED -DMODULE_TGA_ENABLED -DMODULE_THEKLA_UNWRAP_ENABLED -Icore -Icore/math -Ieditor -Idrivers -I. -Iplatform/windows -Ithirdparty/zstd -Ithirdparty/zstd/common -Ithirdparty/zlib -Ithirdparty/rtaudio -Ithirdparty/glad -Ithirdparty/thekla_atlas -Ithirdparty/thekla_atlas/poshlib -Ithirdparty/thekla_atlas/nvcore -Ithirdparty/thekla_atlas/nvmesh thirdparty/thekla_atlas/nvcore/FileSystem.cpp
thirdparty/thekla_atlas/nvcore/Debug.cpp: In function 'int {anonymous}::backtrace(void**, int)':
thirdparty/thekla_atlas/nvcore/Debug.cpp:399:9: error: '_asm' was not declared in this scope
         _asm {
         ^~~~
In file included from thirdparty/thekla_atlas/nvcore/StrLib.cpp:3:0:
thirdparty/thekla_atlas/nvcore/StrLib.h:170:21: note: only here as a friend
         friend void swap(StringBuilder & a, StringBuilder & b);
                     ^~~~
thirdparty/thekla_atlas/nvcore/StrLib.h:359:21: note: only here as a friend
         friend void swap(String & a, String & b);
                     ^~~~
scons: *** [thirdparty/thekla_atlas/nvcore/Debug.windows.opt.tools.32.o] Error 1
```

Compiled on Digitecnology Server for bringing Godot 3 builds. Commit id: 433cb6f49

--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -269,7 +269,7 @@ def configure(env):
 
         if (env["openmp"]):
             env.Append(CPPFLAGS=['-fopenmp'])
-            env.Append(LIBS=['gomp'])
+            env.Append(LINKFLAGS=['-fopenmp'])
 
         ## Compile flags
 

--- a/thirdparty/thekla_atlas/nvcore/Debug.cpp
+++ b/thirdparty/thekla_atlas/nvcore/Debug.cpp
@@ -394,8 +394,10 @@ namespace
 #pragma warning(disable:4748)
     static NV_NOINLINE int backtrace(void * trace[], int maxcount) {
         CONTEXT ctx = { 0 };
+// -- GODOT start --
 #if NV_CPU_X86 && !NV_CPU_X86_64
         ctx.ContextFlags = CONTEXT_CONTROL;
+#if NV_CC_MSVC
         _asm {
              call x
           x: pop eax
@@ -403,6 +405,13 @@ namespace
              mov ctx.Ebp, ebp
              mov ctx.Esp, esp
         }
+#else
+        register long unsigned int ebp asm("ebp");
+        ctx.Eip = (DWORD) __builtin_return_address(0);
+        ctx.Ebp = ebp;
+        ctx.Esp = (DWORD) __builtin_frame_address(0);
+#endif
+// -- GODOT end --
 #else
         RtlCaptureContext(&ctx); // Not implemented correctly in x86.
 #endif

